compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2320,16 +2320,39 @@ private void defineFieldsOfObjectOrRecordTypeDef(BLangTypeDefinition typeDef, Sy
             return;
         }
 
-        if (recordTypeNode.restFieldType == null) {
-            if (recordTypeNode.sealed) {
+        if (recordTypeNode.restFieldType != null) {
+            recordType.restFieldType = symResolver.resolveTypeNode(recordTypeNode.restFieldType, typeDefEnv);
+            return;
+        }
+
+        if (!recordTypeNode.sealed) {
+            recordType.restFieldType = symTable.anydataType;
+            return;
+        }
+
+        for (BLangType typeRef : recordTypeNode.typeRefs) {
+            if (typeRef.type.tag != TypeTags.RECORD) {
+                continue;
+            }
+            BType restFieldType = ((BRecordType) typeRef.type).restFieldType;
+            if (restFieldType == null || restFieldType == symTable.noType) {
+                continue;
+            }
+            if (recordType.restFieldType != null && recordType.restFieldType.tag != restFieldType.tag) {
```

Comment:
Just looking at the tag is not enough, right? For example we can be comparing two different record types here. Need to check if they are the same type.

Reply:
Fixed